server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1231,6 +1235,10 @@ public boolean isRemoteStoreEnabled() {
         return isRemoteStoreEnabled;
     }
 
```

Comment:
The function does not aptly fit in this class

Reply:
agreed . I added it as many classes like `EngineConfig, `Store` and `Translog` have index settings available and they need to find out the same. The alternate would be a good amount of refactor to pass this information from `IndexShard` via some supplier .

Reply:
Node settings only changes on restarts, hence it won't return different values for the same index